onnx/shape_inference/implementation.cc

```diff
@@ -336,7 +336,9 @@ void InferShapeForFunctionNode(
     for (auto attr : n.attribute()) {
       if (attr.has_ref_attr_name()) {
         if (attr_map.count(attr.ref_attr_name())) {
-          copy_n.add_attribute()->CopyFrom(*attr_map[attr.ref_attr_name()]);
+          auto copy_attr = *attr_map[attr.ref_attr_name()];
+          copy_attr.set_name(attr.name());
```

Comment:
We really need to rename attribute when `ref_name != name`.

Reply:
Good catch!